streams/test-utils/src/main/java/org/apache/kafka/streams/TopologyTestDriver.java

```diff
@@ -390,13 +385,10 @@ public void pipeInput(final ConsumerRecord<byte[], byte[]> consumerRecord) {
                 consumerRecord.headers())));
 
             // Process the record ...
-            ((InternalProcessorContext) task.context()).setRecordContext(
```

Comment:
Why remove this `setRecordContext` call?

Reply:
We don't need it because the tasks sets the record context internally anyway. I think it was never required and guess we just c&p from the old internal driver that also made this call (maybe it was required for the older driver or in an older release but never removed -- not sure).